client/my-sites/theme/main.jsx

```diff
@@ -71,48 +67,17 @@ const ThemeSheet = React.createClass( {
 	},
 
 	getInitialState() {
-		return { selectedAction: null };
+		return {
+			showPreview: false,
+		};
 	},
 
 	componentDidMount() {
 		window.scroll( 0, 0 );
 	},
 
-	hideSiteSelectorModal() {
-		this.setState( { selectedAction: null } );
-	},
-
-	isActive() {
-		const { id, currentTheme } = this.props;
-		return currentTheme && currentTheme.id === id;
-	},
-
-	onPrimaryClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else if ( this.isActive() ) {
-			this.props.customize( this.props, this.props.selectedSite );
-		} else if ( this.props.price ) {
-			this.selectSiteAndDispatch( 'purchase' );
-		} else {
-			this.selectSiteAndDispatch( 'activate' );
-		}
-	},
-
-	onPreviewButtonClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else {
-			this.selectSiteAndDispatch( 'customize' );
-		}
-	},
-
-	selectSiteAndDispatch( action ) {
-		if ( this.props.selectedSite ) {
-			this.props[ action ]( this.props, this.props.selectedSite, 'showcase-sheet' );
-		} else {
-			this.setState( { selectedAction: action } );
-		}
+	getTheme() {
+		return pick( this.props, [ 'id', 'name', 'author', 'screenshot', 'price', 'description', 'descriptionLong', 'supportDocumentation', 'download', 'taxonomies', 'stylesheet', 'active', 'purchased' ] );
```

Comment:
Yeah, we've struggled with this in a number of places. It would be great to have a common `theme` object. One obstacle to that is that different endpoints return different fields. 

> imprecise to just pass this.props along since that doesn't describe what the component needs.

I don't see a huge problem with passing `this.props`. The proptypes in the component can describe the needs. I admit it can be confusing at the site where the component is used, though, so your way is fine.